command/agent/alloc_endpoint_test.go

```diff
@@ -1121,3 +1121,29 @@ func TestHTTP_ReadWsHandshake(t *testing.T) {
 		})
 	}
 }
+
+// TestHTTP_AllocPort_Parsing tests that removing the mapstructure tags from the
+// Port struct has no adverse affects on serialization.
+func TestHTTP_AllocPort_Parsing(t *testing.T) {
```

Comment:
I think this test only verifies that the serialized object can still be round-tripped, not that we haven't changed the API? It's probably worth marking this test with a TODO to tear the test out once it's shipped, as it only verifies backcompat.

Reply:
I think the [job_endpoint tests](https://github.com/hashicorp/nomad/blob/24e01efa843b4babe922cfc2007de58c514c8424/command/agent/job_endpoint_test.go#L2650) already check that we're not breaking the API.

Reply:
Ok, and a quick smoke test confirms that:

```
$ nomad operator api "/v1/allocation/73105ebb-22a0-e7aa-621e-cbe72eb31220" | jq '.AllocatedResources.Shared.Networks'
[
  {
    "CIDR": "",
    "DNS": null,
    "Device": "",
    "DynamicPorts": [
      {
        "HostNetwork": "default",
        "Label": "www",
        "To": 8001,
        "Value": 31440
      }
    ],
    "IP": "127.0.0.1",
    "MBits": 0,
    "Mode": "bridge",
    "ReservedPorts": [
      {
        "HostNetwork": "private",
        "Label": "testing",
        "To": 8002,
        "Value": 9002
      }
    ]
  }
]
$ nomad operator api "/v1/allocation/73105ebb-22a0-e7aa-621e-cbe72eb31220" | jq '.SharedResources.Networks'
[
  {
    "CIDR": "",
    "DNS": null,
    "Device": "",
    "DynamicPorts": [
      {
        "HostNetwork": "default",
        "Label": "www",
        "To": 8001,
        "Value": 31440
      }
    ],
    "IP": "127.0.0.1",
    "MBits": 0,
    "Mode": "bridge",
    "ReservedPorts": [
      {
        "HostNetwork": "private",
        "Label": "testing",
        "To": 8002,
        "Value": 9002
      }
    ]
  }
]

```